pageserver/src/tenant/mgr.rs

```diff
@@ -293,14 +293,14 @@ pub async fn set_new_tenant_config(
     info!("configuring tenant {tenant_id}");
     let tenant = get_tenant(tenant_id, true).await?;
 
-    tenant.set_new_tenant_config(new_tenant_conf);
     let tenant_config_path = conf.tenant_config_path(tenant_id);
     Tenant::persist_tenant_config(
         &tenant.tenant_id(),
         &tenant_config_path,
         new_tenant_conf,
         false,
     )?;
+    tenant.set_new_tenant_config(new_tenant_conf);
```

Comment:
> Could also add a fixes https://github.com/neondatabase/neon/issues/3472 to description.

That I've added to the 2nd bullet of the issue description.

> Should we .. 

I think, for dumping an endpoint, it's great that we now do it correctly and keep on-disk and in-memory states in sync.
Seems that initially, it was planned to be a replacing operation, since the config comes from the control plane (cloud, other oracle) that knows what is should be.
So, not sure if it's worth covering safe updates of a small file for this case, since I don't know if that's all needed by the control plane.

Reply:
>> Could also add a fixes #3472 to description.
>
> That I've added to the 2nd bullet of the issue description.

Note: #3472 != #3471, I created two issues from that discussion from whatever PR it was :)

> Seems that initially, it was planned to be a replacing operation, since the config comes from the control plane (cloud, other oracle) that knows what is should be.

Agreed, PUT would also hint at this.

> So, not sure if it's worth covering safe updates of a small file for this case, since I don't know if that's all needed by the control plane.

So the disk full situations would be emergencies if they happened on production, pageserver would be locked on evicting emphemeral files and seemingly livelocked. However in one staging event pitr_cutoff was lowered to free some space (temporarily). Maybe it required other things as well, like removal of the walredo tmp directory.

Because it was config update was attempted as one of the last straws, I think it might be good if it could be updated even in disk full situation..

Reply:
I guess we can implement it separately, at least I am not convinced still, to me, in disk full situations it could be better to evict its layer or somewhow otherwise (detach?) more generally to solve the problem, then trying to write a code to safely update the file.

Reply:
I reached out to the control plane team about this.

Reply:
[Internal Slack thread with control plane team](https://neondb.slack.com/archives/C03438W3FLZ/p1675432826659139)

Reply:
@koivunej was the config changes through the API endpoint, or by modifying the tenant's toml on disk and restarting the pageserver.

My gut feeling is that in practice, if we run out of disk, there's an outage and pageserver.service will likely be restarted by the operator after making free space.

Reply:
I think I remember seeing comments about changing it's pitr setting while at full disk, but didn't look this up. It definitetly was not done through console.

But yeah, perhaps this we should just make some internal announcements about this, "don't try this anymore" :)

At the same time, just removing a temp walredo dir might help with the 32MB...

Reply:
Ok, so moving on and keeping things as is, if I've understood the discussion right.